bgpd/rfapi/rfapi_vty.c

```diff
@@ -1174,16 +1174,18 @@ rfapiPrintRemoteRegBi (
                 inet_ntop (pfx_un.family, &pfx_un.u.prefix, buf_ntop,
                            BUFSIZ));
     }
-  buf_un[BUFSIZ - 1] = 0;
 
   rfapiGetTunnelType(bi->attr,&tun_type);
   /*
    * VN addr
    */
   buf_vn[0] = 0;
+  rfapiNexthop2Prefix (bi->attr, &pfx_vn);
   if (tun_type == BGP_ENCAP_TYPE_MPLS)
     {
       /* MPLS carries un in nrli next hop (same as vn for IP tunnels) */
+      snprintf (buf_un, BUFSIZ, "%s",
+                inet_ntop (pfx_vn.family, &pfx_vn.u.prefix, buf_ntop, BUFSIZ));
```

Comment:
We need to bulk-fix all parameter uses of `BUFSIZ` - these should really be sizeof(buf). We already had one really bad security vulnerability due to this, let's not repeat the mistake :/

(I think I can do this with a coccinelle semantic patch script)

Merging anyway, it's correct in this location and all over the code anyway :(